### DISCUSSION TITLE
Ansible Errors During Installation

### DISCUSSION THREAD

**Discussion by ph0nebooth** [hnimminh/libresbc#66] [Q&A] [answered]
Hi everyone

I'm new to Ansible with LibreSBC being the first app I've deployed. I've faced many errors during the deployment on a clean Debian 10 VM, most of which I have overcome with some basic troubleshooting and research. However, this error I can't overcome.

TASK [libre : install lua pkgs] ****************************************************************************************************************************************************************************
fatal: [sbc]: FAILED! => {"changed": false, "cmd": "/bin/sh requirement.sh", "delta": null, "end": null, "msg": "Unable to change directory before execution: [Errno 2] No such file or directory: b'/opt/libresbc/0.5.9/callng'", "rc": null, "start": null, "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

The /opt/libresbc/0.5.9 directory only contains the folder 'liberator' and it's empty.

All tasks prior to 'install lua pkgs' were successful. Here is a snippet of the two tasks which I believe are related.

TASK [libre : Create directory /opt/libresbc/0.5.9/callng in target host] **********************************************************************************************************************************
--- before
+++ after
@@ -1,4 +1,4 @@
 {
     "path": "/opt/libresbc/0.5.9/callng",
-    "state": "absent"
+    "state": "directory"
 }

changed: [sbc]

TASK [libre : Copy libre role source to target host] *******************************************************************************************************************************************************
root@203.29.X.X's password: 
cd+++++++ ./
<f+++++++ callfunc.lua
<f+++++++ configuration.lua
<f+++++++ event.capacity.lua
<f+++++++ event.cdr.lua
<f+++++++ event.startup.lua
<f+++++++ event.unauth.lua
<f+++++++ kami.lua
<f+++++++ main.lua
<f+++++++ redirection.lua
<f+++++++ requirement.sh
<f+++++++ sigfunc.lua
<f+++++++ utilities.lua
changed: [sbc]

Please let me know if I can provide any further logs. Like I mentioned, I'm new to Ansible so I'm not sure if this information is what the community requires.

Thanks

EDIT

I worked around the error by manually cloning the files to the directory as appropriate. The next error I ran into was the captagent-6.3.1.tar.gz file doesn't exist. It appears there is an issue with files being pulled/downloaded via the Ansible 'playbook'.

**Comment by hnimminh** [comment]
look like source was not copy to destination host, can you share a full ansible log? @ph0nebooth

**Reply by ph0nebooth** [reply]
Hi Minh

Thank you for your reply. I noticed that there is a shell script for installation in the build directory. I'm trying that install method then I'll report back. If that also fails I'll send through the Ansible log.

Thank you

**Reply by ph0nebooth** [reply]
> look like source was not copy to destination host, can you share a full ansible log? @ph0nebooth

Hi @hnimminh 

The shell script installation didn't work. I was unable to browse to :8443/apidocs at the end of the installation.

Here is a copy of the Ansible log. It's now failing when trying to obtain the FreeSWITCH GPG key.

[ansible.log](https://github.com/hnimminh/libresbc/files/9486551/ansible.log)

**Reply by hnimminh** [reply]
`The shell script installation didn't work` it is expectation. shell script is a installation noted. The document has instructed installation done via ansible only. https://github.com/hnimminh/libresbc/wiki/Installation

FreeSWITCH GPG is required, you can get more detail from https://github.com/hnimminh/libresbc/issues/51

**Reply by ph0nebooth** [reply]
> `The shell script installation didn't work` it is expectation. shell script is a installation noted. The document has instructed installation done via ansible only. https://github.com/hnimminh/libresbc/wiki/Installation
> 
> FreeSWITCH GPG is required, you can get more detail from #51

I understand. I have the signalwire_pat_token token defined in the Ansible command and this wasn't an issue prior to today. 

ansible-playbook playbooks/deployment.yml -i inventories/production -l "sbc" -t "platform,libre,nginx,captagent" -e "signalwire_pat_token=pat_XXXXXXXXXXXXXXXXXX"--check -k -vvv

**Reply by hnimminh** [reply]
there is nothing change on libresbc today, it may be an issue of signalwise.

**Reply by ph0nebooth** [reply, marked as answer]
> there is nothing change on libresbc today, it may be an issue of signalwise.

I manually downloaded and installed the GPG key using the username "signalwire" and my PAT. See https://freeswitch.org/confluence/display/FREESWITCH/Debian. I noticed in your freeswitch.yml role you use the username "libresbc". Is there something wrong with that account, perhaps?

I ran the Ansible playbook again and LibreSBC has been successfully installed. I can access the API Docs.

I look forward to test out LibreSBC!

**Reply by MikeQc99** [reply]
We encountered the same issue. Could the error come from a change from the Ansible get_url module ? It seems like user and password are not accepted anymore with this module. I tried to use the token only with client_key parameter instead but it return a authentication error.

Her the error
`fatal: [Cleopatra]: FAILED! => {"changed": false, "msg": "Unsupported parameters for (get_url) module: password, username Supported parameters include: attributes, backup, checksum, client_cert, client_key, content, delimiter, dest, directory_mode, follow, force, force_basic_auth, group, headers, http_agent, mode, owner, regexp, remote_src, selevel, serole, setype, seuser, sha256sum, src, timeout, tmp_dest, unsafe_writes, url, url_password, url_username, use_proxy, validate_certs"}`

**Comment by netstudy-2024** [comment]
I have also encountered this problem （"Unsupported parameters for (get_url) module:） now. May I ask how to solve it?
